aeron-driver/src/main/c/concurrent/aeron_thread.c

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2014-2019 Real Logic Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "concurrent/aeron_thread.h"
+
+
+#if defined(AERON_COMPILER_GCC)
+
+#elif defined(AERON_COMPILER_MSVC) && defined(AERON_CPU_X64)
+
+void pthread_once(AERON_INIT_ONCE* s_init_once, void* callback)
```

Comment:
I'm not sure I understand in which scenario this would be a problem for:
If you are building for cygwin, wouldn't you use gcc/clang ? (in which case, AERON_COMPILER_MSVC would not be defined)

To support this, we'll need to create an indrection for all that code and the shared/linux code will be impacted too. Is that what you mean/want ?

Reply:
C media driver can be used as a library and brought into anything as a binary dep.

Reply:
An inline function can reduce the impact for those that are trivial. Some of them are different enough that it isn't a straight "make a version for Windows with the same name". So, refactoring what is being done might be more appropriate.